proposals/0017-conforming-literals.md

```diff
@@ -172,6 +172,32 @@ important note about this title is that it is an Xbox & Windows exclusive title.
 Since the title is platform exclusive its shaders may not need to be resilient
 to different language semantics implemented by different shader compilers.
 
+#### Rendering Defect Root Cause
+
+Continued investigation of the rendering defects tracked in
+[#184](https://github.com/microsoft/hlsl-specs/issues/184) revealed that the
+problems in both the catastrophic and subtle cases are not caused by loss of
+floating point precision. Instead they are the result of a subtle behavioral
+difference for literal integers. In HLSL `literal int` is always a signed
+integer, whereas in C a non-base 10 literal integer may be unsigned if the most
+significant bit is 1.
+
+This difference in rules results in a subtle behavioral difference for bit
+shifts. Given the following code:
+
+```c++
+export float Fn(int inInt, inFloat) {
+  int Val = (inInt & 0xffff0000) >> 16);

```

Comment:
Given that treating literals as unsigned caused all of the rendering regressions I identified in my testing, I don't think we can change this behavior without a language version. Applying this retroactively will break existing shaders and result in the catastrophic rendering defects that were identified during this testing.

I'm going to capture the FXC behavior in this document so that we don't lose it.

Reply:
I've captured the notes here in the proposal.